components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/util/ActionExecutorConfig.java

```diff
@@ -73,6 +80,56 @@ public boolean isExecutionForActionTypeEnabled(ActionType actionType) {
         }
     }
 
+    /**
+     * Retrieves the HTTP read timeout configuration.
+     * If the configuration value is invalid or missing, the default timeout value is parsed.
+     *
+     * @return The HTTP read timeout int value in milliseconds.
+     */
+    public int getHttpReadTimeout() {
+
+        return parseTimeoutConfig(HTTP_READ_TIMEOUT_PROPERTY, DEFAULT_HTTP_READ_TIMEOUT);
+    }
+
+    /**
+     * Retrieves the HTTP connection request timeout configuration.
+     * If the configuration value is invalid or missing, the default timeout value is parsed.
+     *
+     * @return The HTTP connection request timeout int value in milliseconds.
+     */
+    public int getHttpConnectionRequestTimeout() {
+
+        return parseTimeoutConfig(HTTP_CONNECTION_REQUEST_TIMEOUT_PROPERTY, DEFAULT_HTTP_CONNECTION_REQUEST_TIMEOUT);
+    }
+
+    /**
+     * Retrieves the HTTP connection timeout configuration.
+     * If the configuration value is invalid or missing, the default timeout value is parsed.
+     *
+     * @return The HTTP connection timeout int value in milliseconds.
+     */
+    public int getHttpConnectionTimeout() {
+
+        return parseTimeoutConfig(HTTP_CONNECTION_TIMEOUT_PROPERTY, DEFAULT_HTTP_CONNECTION_TIMEOUT);
+    }
```

Comment:
Let's denote that each is milliseconds in the method name

Reply:
Updated in https://github.com/wso2/carbon-identity-framework/pull/5935/commits/5f6e873c091b324a61192f2e25555d1675e2934c